### DISCUSSION TITLE
Help with nextjs overrided output directory

### DISCUSSION THREAD

**Discussion by shadow3312** [vercel/community#7285] [Help] [answered]
### Summary

I'm trying to deploy a nextjs project inside a monorepo to vercel for the first time. My project repo contains two folders. server and web. The web folder contains the nextjs app.  I cannot set Root directory to /web and select the Nextjs preset since my nextjs project is using tRPC to communicate with the backend, so the build will fail since the root dir cannot access anything from outside. I had to manually set the output directory, but it gives a 404 when visiting the url after successful build. Hope I could get some tips. Thanks.

### Example

https://zbanking.vercel.app

### Steps to Reproduce

The repo can be found here, including READMEs to reproduce locally [https://github.com/shadow3312/banking_fs](url)

Everything works perfectly in local (dev and build)

The obviously problematic vercel config: 
![Capture d'écran 2024-06-26 031000](https://github.com/vercel/community/assets/30819762/45ae2bca-ecb3-4e12-a547-2c7c2a548784)

![Capture d'écran 2024-06-26 031026](https://github.com/vercel/community/assets/30819762/35675caf-0929-467f-8139-ab567da10b50)

And I'm pretty sure the vercel build output is not supposed to be this

![Capture d'écran 2024-06-26 032139](https://github.com/vercel/community/assets/30819762/77363196-5e8b-41fb-bb2f-8cff2703ffa6)

**Comment by pawlean** [comment]
Hi @shadow3312! 

> [https://github.com/shadow3312/banking_fs](https://github.com/orgs/vercel/discussions/url)
This is private by the way. 

But from what you described and the screenshots you've shown, it looks correct. Can you try setting the **Root Directory** to `web` as this is where the application code lives?

Let me know how you get on!

**Reply by shadow3312** [reply]
Thanks for your answer.

> This is private by the way.
This is weird
![Capture d'écran 2024-06-26 161732](https://github.com/vercel/community/assets/30819762/89d31389-764d-45bd-b9db-633b38554719)

Can you try to just copy/paste the link in the browser instead of clicking the generated one here ?

Here is the updated setup
![Capture d'écran 2024-06-26 163800](https://github.com/vercel/community/assets/30819762/93df1e83-3041-4e33-94cf-518c09a4483b)

![Capture d'écran 2024-06-26 163739](https://github.com/vercel/community/assets/30819762/20768e52-2174-44ce-9fed-804ff3db95d2)



This setup build is successful but I still get the 404. 

I then tried to select the Nextjs framework preset, but here the build fail




![Capture d'écran 2024-06-26 164508](https://github.com/vercel/community/assets/30819762/224f8844-cb83-49d8-af45-fed0b2747b2d)

**Comment by shadow3312** [comment, marked as answer]
I figured it out, Basically, since the NODE_ENV is set to production in vercel, `npm install` skips the devDependencies. Adding `npm install --include dev` solves the issue.

**Reply by pawlean** [reply]
Nice! Thank you for coming back and sharing the solution with the community. Let us know if we can help with anything else!

[a comment by an automated account is not shown]
